### DISCUSSION TITLE
Troubleshooting Scrapli Connectivity Issue with Ubuntu Hosts

### DISCUSSION THREAD

**Discussion by ssmyk** [carlmontanari/scrapli#325] [Q&A]
Hello,
I have a problem with the latest version of Scrapli. When I am trying to connect to an Ubuntu host, the communication freezes, and a timeout is triggered:
```
DEBUG:scrapli.channel:read: b'ping 8.8.8.8 -c 1'
DEBUG:scrapli.channel:write: '\n'
DEBUG:scrapli.channel:read: b'\n'
DEBUG:scrapli.channel:read: b'\x1b[?2004h\x1b]0;user@line5-cpe-0: ~\x07user@line5-cpe-0:~$ ping 8.8.8.8 -c 1\n\x1b[?2004l'
DEBUG:scrapli.channel:read: b'PING 8.8.8.8 (8.8.8.8) 56(84) bytes of data.\n'
DEBUG:scrapli.channel:read: b'64 bytes from 8.8.8.8: icmp_seq=1 ttl=117 time=21.2 ms\n\n--- 8.8.8.8 ping statistics ---\n1 packets transmitted, 1 received, 0% packet loss, time 0ms\nrtt min/avg/max/mdev = 21.162/21.162/21.162/0.000 ms\n'
DEBUG:scrapli.channel:read: b'\x1b[?2004h\x1b]0;user@line5-cpe-0: ~\x07user@line5-cpe-0:~$ '
CRITICAL:scrapli.channel:operation timed out, closing connection
```
It works correctly in version 2023.7.30. I think it's this change that's causing the problem:
https://github.com/carlmontanari/scrapli/pull/316/files

**Comment by carlmontanari** [comment]
hey @ssmyk would you be able to mess around and try to see if you can get that pattern working with your setup + the existing tests?

**Reply by ssmyk** [reply]
Yes, I found solution for me, there is a summary:
I compare how regex works with my ssh server output:
2023.7.30: `[(b'[?2004h', b''), (b']0;user@line5-cpe-0: ~\x07', b'\x07')]`
2024.1.30: `[(b'', b'[?2004h', b'', b'', b'[?2004h')]`
So it seems the match with `\x07` is missing in the newest version. This change in `ANSI_ESCAPE_PATTERN` fixes it:
`rb"([\x07])" -> rb"(.*[\x07])"` and then i have match:
`[(b'', b'[?2004h\x1b]0;user@line5-cpe-0: ~\x07', b'', b'[?2004h\x1b]0;user@line5-cpe-0: ~\x07', b'')]`
or to be more precise:
`rb"([\x07])" -> rb"((\]\d).*[\x07])"` which provides to: 
`[(b'', b'[?2004h', b'', b'', b'', b'[?2004h'), (b'', b']0;user@line5-cpe-0: ~\x07', b'', b']0;user@line5-cpe-0: ~\x07', b']0', b'')]`

**Reply by carlmontanari** [reply]
thanks so much @ssmyk can you check out #326 and make sure that sorts things out for ya?

**Reply by ssmyk** [reply]
Yes, it seems it fixes the issue.
